### PR TITLE
Improved testing: Reducers

### PR DESCRIPTION
**Pull request by wwqrd**
This PR only has reducers tests for the protocol and error modules because I'm moving onto the Architect project.

Includes a working example for testing epics (rxjs/redux observables).

Partial: #386

### CODE REVIEW

File: src/ducks/modules/errors.js [hunk 1]
```diff
@@ -16,7 +14,7 @@ export default function reducer(state = initialState, action = {}) {
     case errorActionTypes.DOWNLOAD_PROTOCOL_FAILED:
     case errorActionTypes.IMPORT_PROTOCOL_FAILED:
     case errorActionTypes.LOAD_PROTOCOL_FAILED:
-      console.error(action.error); // eslint-disable-line
+      if (!jest) { console.error(action.error); } // eslint-disable-line
```
[the remark this answers is not in the document]
Comment (bryfox) [reply src/ducks/modules/errors.js]: Or instead mock console.error in that test file (`console.error = jest.fn()`)?

File: src/ducks/modules/__tests__/protocol.test.js [hunk 2]
```diff
@@ -0,0 +1,163 @@
+/* eslint-env jest */
+
+import { ActionsObservable } from 'redux-observable';
+import 'rxjs';
```
Comment (bryfox) [inline src/ducks/modules/__tests__/protocol.test.js]: Is this import needed?

Comment (wwqrd) [reply src/ducks/modules/__tests__/protocol.test.js]: No it isn't! Have removed it.

File: src/ducks/modules/errors.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (wwqrd) [reply src/ducks/modules/errors.js]: I think that's a good call, I've mocked `console.error` in the jest setup.

**Comment by rebeccamadsen** [comment]
Looks good to me too.

**Pull request merged**